### PR TITLE
Add `ignorePathResolveFrom` option for reusable ignores

### PR DESCRIPTION
**Pull request by wooorm**
Closes GH-41.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -125,6 +125,8 @@ done.
     — Filepath to an ignore file to load
 *   [`ignorePatterns`][ignore-patterns] (`Array.<string>`, optional)
     — Patterns to ignore in addition to ignore files, if any
+*   [`ignoreFrom`][ignore-from] (`string`, optional)
+    — Path to directory to resolve ignore patterns from
```
Comment (ChristianMurphy) [inline readme.md]: I take it the default is `cwd`?
Is that documented?

Comment (wooorm) [reply readme.md]: The default depends on whether the file is given as a path (in which case, its directory is used) or if patterns are directly given (in which case, cwd).

I described the defaults in `docs/ignore` and `docs/options` in this PR too!

Comment (ChristianMurphy) [reply readme.md]: Gotcha, thanks

**Comment by wooorm** [comment]
> Allowing ignores to be in unrelated and apply to unrelated folders breaks feels like it could cause as much confusion and frustration for non-power users as it would resolve frustration for power users.

I agree that it is a power user, and pretty confusing, option. I’d love to pick a better solution, just not sure how that would look!

> Because the file isn't part of the repo, or locked in a package.json, it could be different for each user.

Your `/etc` example could be used indeed but it probably doesn’t make a lot of sense for an open source project. The engine can use exactly one ignore file, so if there was `cli --ignore-path /etc/ignore` in the project, then it would cause an exception for users who don’t have that file.

[Athan’s example](https://github.com/unifiedjs/unified-engine/issues/41#issue-497290845) is `cli --rc-path=./etc/remark/.remarkrc.js --ignore-path=./etc/remark/.remarkignore files and globs`. The configs are in [`etc/remark`](https://github.com/stdlib-js/stdlib/tree/develop/etc/remark) because it’s a big project with otherwise so many dotfiles in the project root.

A different [use case from Alex](https://github.com/unifiedjs/unified-engine/issues/41#issuecomment-548807253) is `remark --ignore-path=node_modules/my-package/remarkignore`.

I tried to boil down the use cases [here](https://github.com/unifiedjs/unified-engine/issues/41#issuecomment-548819258).

So while your `/etc` example indeed is not great (because the file isn’t in the repo/versioned), the other two examples are!

---

One more solution: `ignorePatternsPath` option to load a file with ignore patterns that are resolved from cwd.

Then the options are (in addition to automatically loaded ignore files):
- `ignorePath` — Path to pattern file resolved whose patterns are resolved from the path’s dirname
- `ignorePatterns` — List of patterns resolved from $cwd
- `ignorePatternsPath` — Path to pattern file whose patterns are resolved from $cwd

One more alternative: Maybe `cli --ignorePatterns="$(cat ./etc/ignore)"` already works?

---

> Since the goal it to make ignores reusable, might it be possible to make ignores something that could be included in a Preset? so that it could be shared in a versioned way?

Hmm, presets are loaded from config files, which are loaded based on files that were found (and thus not ignored). That gets tricky! 😬

**Comment by ChristianMurphy** [comment]
> Your `/etc` example could be used indeed but it probably doesn’t make a lot of sense for an open source project.

I completely agree.
My point is less that many project would or should do something like that,
More that without clear warnings that `ignoreFrom` shouldn't be used in most cases, that people will try to use it not fully understanding its implications, resulting in confusion and frustration.

A partial solution for the current approach could be limiting this option from going above `cwd`, limiting it's ability to reach out of the project context.
Which would limit confusion on linting parent folders, but could still have confusion around the round about nature of the relative resolves.

> One more solution: `ignorePatternsPath` option to load a file with ignore patterns that are resolved from cwd.

I think this could be clearer.

> One more alternative: Maybe `cli --ignore-pattern="$(cat ./etc/ignore)"` already works?

Even better!
Allow the shell to do the heavy lifting.

---

One more possible solution.
An `ignoreRelativeTo` option, with possible values of `'ignore-file'` and `'cwd'`.

**Comment by wooorm** [comment]
> A partial solution for the current approach could be limiting this option from going above cwd, limiting it's ability to reach out of the project context.

Downside of that is, say you’re in `tests/`, then doing `../node_modules/remark-cli/cli ...` wouldn‘t work. It’s not a very important case, but that’s what I’ve been thinking about too. My other proposed solutions don’t handle this either though.

**Comment by wooorm** [comment]
Went with your last suggestion, with a slightly more verbose name:

`ignorePathResolveFrom`, which can be set to `dir` or `cwd`, to resolve from the ignore file’s directory, or the current working directory.

**Pull request merged**